### DISCUSSION TITLE
Arrows in Excercise

### DISCUSSION THREAD

**Discussion by akaKarthick** [czbar/ChessForge#630] [General]
Can we hide the arrows from the excercise. But dont restrict to draw arrows in solving mode.

https://youtu.be/gBSszfyjdpI

**Comment by czbar** [comment]
Yes, this was actually the intent but not properly implemented. Fixed in 1.1.4 (now available from https://chessforge.sourceforge.io/Releases/latest/ChessForge.msi)

**Comment by akaKarthick** [comment]
But now its not allowing me to draw arrows in solving mode.

**Comment by czbar** [comment]
Yes, I have disabled both viewing and drawing in the solving mode.  I thought that's what we wanted, no?

I can allow drawing in the Solving mode but then what would the purpose be?  User's solutions are not saved so the drawings would not be persisted after exiting the solving mode.  Or do you think that it would help the user to solve the exercise if they are allowed to draw?

Let me know what scenario, or use case, you have in mind for this.

**Comment by akaKarthick** [comment]
I am creating chess challenges for my niece. When I create an exercise, I draw arrows for my reference in editing mode. Most of the time, I delete these arrows, but sometimes I forget. When my niece solves the exercise in solving mode, she can see my reference arrows(1.1.3.0). 

**I would like to hide or delete these arrows so that she can only see the arrows that she draws herself.** 

Now i can draw the arrows in exercise (editing mode). And that arrows are not visible in solving mode. That's good

but she can't draw the arrows. (solving mode 1.1.4.0). 

Expected behaviour: **able to draw shapes in solving mode.**

See. I demonstrate the issue here.
[https://youtu.be/gBSszfyjdpI](https://youtu.be/gBSszfyjdpI)

This is my scenario. (#615 , #621 related to this scenario)
if it doesnt make sense lets ignore this. if this is an issue lets raise new issue.

**Comment by czbar** [comment]
It's just that I am not sure if the person solving an exercise does want the ability to draw the arrows. I can enable it but then they won't be saved anywhere. Wouldn't that be confusing? I am honestly not sure.

I guess, we should ask the users. What does your niece say? Does she miss the ability to draw arrows while solving?

**Reply by akaKarthick** [reply]
She just asks for it. She is not old enough to think deeply and remember. She is only 10 years old. 

Let's ask another user if we want. 

If it doesn't seem to be needed, we will leave it and fix the other bugs. We'll be back to this thought later if needed.

**Reply by czbar** [reply]
In the next version (1.2.2) I will allow drawing in the Solving mode.  I guess, there is no harm in it.
